### PR TITLE
Separate cairo-xlib.h Lua bindings from cairo.h

### LINKED ISSUE #1804: [Enhancement]: Lua libraries require in-source builds
In-source builds polute the working directory and should be avoided. Conky handles out-of-source building correctly for the most part, but there's an issue when trying to build conky with `BUILD_LUA_CAIRO` and/or other Lua libraries.

When trying to `make install` conky from a nested `./build` directory, it fails to find built lua libraries (e.g. cairo):
<details>
<summary><code>make install</code> log</summary>
<pre>
$ sudo make install
-- CMAKE_C_FLAGS: 
-- CMAKE_CXX_FLAGS: 
-- CMAKE_C_FLAGS_DEBUG: -g
-- CMAKE_CXX_FLAGS_DEBUG: -g
-- CMAKE_C_FLAGS_MINSIZEREL: -Os -DNDEBUG
-- CMAKE_CXX_FLAGS_MINSIZEREL: -Os -DNDEBUG
-- CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
-- CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
-- CMAKE_C_FLAGS_RELWITHDEBINFO: -O2 -g -DNDEBUG
-- CMAKE_CXX_FLAGS_RELWITHDEBINFO: -O2 -g -DNDEBUG
-- CMAKE_BUILD_TYPE: Debug
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Found Lua: /usr/lib/liblua5.4.so;/usr/lib/libm.so (found suitable version "5.4.6", minimum required is "5.3")
Configuration for target conky-cairo
-- SOURCE_DIR = /data/code/FOSS/conky/lua
-- BINARY_DIR = /data/code/FOSS/conky/build/lua
-- COMPILE_OPTIONS = $<$<COMPILE_LANG_AND_ID:CXX,Clang>:-stdlib=libc++>
-- INCLUDE_DIRECTORIES = /data/code/FOSS/conky/3rdparty/toluapp/include;/usr/include/freetype2;/usr/include;/usr/include/cairo;/usr/include/libpng16;/usr/include/harfbuzz;/usr/include/glib-2.0;/usr/lib/glib-2.0/include;/usr/include/sysprof-6;/usr/include/pixman-1;/usr/include;/data/code/FOSS/conky/lua
-- LINK_LIBRARIES = cairo;X11;Xext;/usr/lib/liblua5.4.so;/usr/lib/libm.so;toluapp_lib_static
Configuration for target conky-imlib2
-- SOURCE_DIR = /data/code/FOSS/conky/lua
-- BINARY_DIR = /data/code/FOSS/conky/build/lua
-- COMPILE_OPTIONS = $<$<COMPILE_LANG_AND_ID:CXX,Clang>:-stdlib=libc++>
-- INCLUDE_DIRECTORIES = /data/code/FOSS/conky/3rdparty/toluapp/include;/usr/include/freetype2;/usr/include;/usr/include/cairo;/usr/include/libpng16;/usr/include/harfbuzz;/usr/include/glib-2.0;/usr/lib/glib-2.0/include;/usr/include/sysprof-6;/usr/include/pixman-1;/usr/include;/data/code/FOSS/conky/lua;/usr/include;/usr/include;/usr/include;/data/code/FOSS/conky/build;/data/code/FOSS/conky/lua;/data/code/FOSS/conky/src
-- LINK_LIBRARIES = -L/usr/lib;-lImlib2;/usr/lib/liblua5.4.so;/usr/lib/libm.so;toluapp_lib_static
Configuration for target conky-rsvg
-- SOURCE_DIR = /data/code/FOSS/conky/lua
-- BINARY_DIR = /data/code/FOSS/conky/build/lua
-- COMPILE_OPTIONS = $<$<COMPILE_LANG_AND_ID:CXX,Clang>:-stdlib=libc++>
-- INCLUDE_DIRECTORIES = /data/code/FOSS/conky/3rdparty/toluapp/include;/usr/include/freetype2;/usr/include;/usr/include/cairo;/usr/include/libpng16;/usr/include/harfbuzz;/usr/include/glib-2.0;/usr/lib/glib-2.0/include;/usr/include/sysprof-6;/usr/include/pixman-1;/usr/include;/data/code/FOSS/conky/lua;/usr/include;/usr/include;/usr/include;/data/code/FOSS/conky/build;/data/code/FOSS/conky/lua;/data/code/FOSS/conky/src;/usr/include/librsvg-2.0;/usr/include/gdk-pixbuf-2.0;/usr/include;/usr/include/cairo;/usr/include/glib-2.0;/usr/lib/glib-2.0/include;/usr/include/libmount;/usr/include/blkid;/usr/include/sysprof-6;/usr/include/freetype2;/usr/include/libpng16;/usr/include/harfbuzz;/usr/include/pixman-1;/usr/include;/data/code/FOSS/conky/lua
-- LINK_LIBRARIES = rsvg-2;m;gio-2.0;glib-2.0;gdk_pixbuf-2.0;gobject-2.0;cairo;/usr/lib/liblua5.4.so;/usr/lib/libm.so;toluapp_lib_static
-- Found Lua: /usr/lib/liblua5.4.so;/usr/lib/libm.so (found version "5.4.6")
Configuration for target conky
-- SOURCE_DIR = /data/code/FOSS/conky/src
-- BINARY_DIR = /data/code/FOSS/conky/build/src
-- COMPILE_OPTIONS = $<$<COMPILE_LANG_AND_ID:CXX,Clang>:-stdlib=libc++>
-- INCLUDE_DIRECTORIES = /data/code/FOSS/conky/3rdparty/toluapp/include;/data/code/FOSS/conky/build;/usr/include;/usr/include;/usr/include/freetype2;/usr/include/freetype2;/usr/include;/usr/include;/usr/include;/usr/include;/data/code/FOSS/conky/build/data
-- LINK_LIBRARIES = -lm;/usr/lib/libncurses.so;/usr/lib/libform.so;/usr/lib/libSM.so;/usr/lib/libICE.so;/usr/lib/libX11.so;/usr/lib/libXext.so;/usr/lib/libXdamage.so;/usr/lib/libXfixes.so;/usr/lib/libXft.so;/usr/lib/libfontconfig.so;/usr/lib/libXext.so;/usr/lib/libXinerama.so;/usr/lib/libXfixes.so;/usr/lib/libXi.so;/usr/lib/libxcb.so;/usr/lib/libxcb.so;/usr/lib/libxcb-errors.so;/usr/lib/liblua5.4.so;/usr/lib/libm.so;-L/usr/lib;-lImlib2;toluapp_lib_static;tcp-portmon
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: /data/code/FOSS/conky/build
[  7%] Built target toluapp_lib_static
[ 11%] Built target toluapp
[ 13%] Built target tcp-portmon
[ 16%] Built target generated_hdr_files
[100%] Built target conky
Install the project...
-- Install configuration: "Debug"
CMake Error at lua/cmake_install.cmake:60 (file):
  file INSTALL cannot find "/data/code/FOSS/conky/build/lua/libcairo.so": No
  such file or directory.
Call Stack (most recent call first):
  cmake_install.cmake:47 (include)
</pre>
</details>

### PR DESCRIPTION
**Pull request by Caellian**
This allows cairo.h to be used from Lua on Wayland without X11 as requirement.

A way of creating surfaces in Lua that uses Shm allocated buffers should be added in future. This PR only cleans up build options to avoid weird errors.

Closes #1804 - `BUILD_LUA_CAIRO` didn't depend on X11 and `lua/CMakeLists.txt` doesn't provide warnings like `dependent_option`s do, so it wasn't clear why the build was failing with `BUILD_LUA_CAIRO && !BUILD_X11`.

### CODE REVIEW

File: .github/workflows/build-and-test-linux.yaml [hunk 1]
```diff
@@ -88,7 +88,6 @@ jobs:
             libaudclient-dev \
             libcairo2-dev \
             libcurl4-gnutls-dev \
-            libcurl4-gnutls-dev \
```
Comment (Caellian) [inline .github/workflows/build-and-test-linux.yaml:91]: Duplicate line?

**Comment by jjgomera** [comment]
if this PR has been included in v1.20.2 included in debian trixie, I still have the issue

`attempt to call a nil value (global 'cairo_xlib_surface_create')`

using the conky rings lua script

**Comment by Caellian** [comment]
@jjgomera In later versions you need to `require cairo_xlib` from Lua, see [here](https://github.com/brndnmtthws/conky/issues/1867#issuecomment-2088429729) for more details. This was separated out to help with future wayland support.

**Comment by jjgomera** [comment]
> @jjgomera In later versions you need to `require cairo_xlib` from Lua, see [here](https://github.com/brndnmtthws/conky/issues/1867#issuecomment-2088429729) for more details. This was separated out to help with future wayland support. Lmk if it still doesn't work with that.

Thanks, thats was the problem, work now!!

**Pull request merged**